pep-3145.txt

```diff
@@ -5,136 +5,150 @@ Last-Modified: $Date$
-    different implementations of Popen.send and Popen._recv have identical
-    arguments to make code that uses these functions work across multiple
-    platforms.  
+Reference Implementation
+========================
 
-    When calling the Popen._recv function, it requires the pipe name be
-    passed as an argument so there exists the Popen.recv function that passes
-    selects stdout as the pipe for Popen._recv by default. Popen.recv_err
-    selects stderr as the pipe by default. Popen.recv and Popen.recv_err
-    are much easier to read and understand than Popen._recv('stdout' ...) and
-    Popen._recv('stderr' ...) respectively.  
+I have been maintaining a Google Code repository that contains all of my
+changes including tests and documentation [9]_ as well as blog detailing
+the problems I have come across in the development process [10]_.
 
-    Since the Popen._recv function does not wait on data to be produced
-    before returning a value, it may return empty bytes. Popen.asyncread
-    handles this issue by returning all data read over a given time
-    interval.  
+I have been working on implementing non-blocking asynchronous I/O in the
+subprocess.Popen module as well as a wrapper class for subprocess.Popen
+that makes it so that an executed process can take the place of a file by
+duplicating all of the methods and attributes that file objects have.
 
-    The ProcessIOWrapper class uses the asyncread and asyncwrite functions to
-    allow a process to act like a file so that there are no blocking issues
-    that can arise from using the stdout and stdin file objects produced from
-    a subprocess.Popen call.
-    
+There are two base functions that have been added to the subprocess.Popen
+class: Popen.send and Popen._recv, each with two separate implementations,
+one for Windows and one for Unix-based systems.  The Windows
+implementation uses ctypes to access the functions needed to control pipes
+in the kernel 32 DLL in an asynchronous manner.  On Unix based systems,
+the Python interface for file control serves the same purpose.  The
+different implementations of Popen.send and Popen._recv have identical
+arguments to make code that uses these functions work across multiple
+platforms.
 
-References:
+When calling the Popen._recv function, it requires the pipe name be
+passed as an argument so there exists the Popen.recv function that passes
+selects stdout as the pipe for Popen._recv by default. Popen.recv_err
+selects stderr as the pipe by default. Popen.recv and Popen.recv_err
+are much easier to read and understand than Popen._recv('stdout' ...) and
+Popen._recv('stderr' ...) respectively.
 
-    [1] [ python-Feature Requests-1191964 ] asynchronous Subprocess
-        http://mail.python.org/pipermail/python-bugs-list/2006-December/
-          036524.html
+Since the Popen._recv function does not wait on data to be produced
+before returning a value, it may return empty bytes. Popen.``asyncread``
```

Comment:
I guess you meant \`\`Popen.asyncread\`\` right? I'd keep the old formatless version since all of the other Popen.* or subprocess.Popen instances are not wrapped by backtics.